Libraries/src/Amazon.Lambda.AspNetCoreServer/APIGatewayHttpApiV2ProxyFunction.cs

```diff
@@ -175,6 +176,18 @@ protected override void MarshallRequest(InvokeFeatures features, APIGatewayHttpA
                 // was marshalled into ASP.NET Core request.
                 PostMarshallConnectionFeature(connectionFeatures, apiGatewayRequest, lambdaContext);
             }
+
+            {
+                var tlsConnectionFeature = (ITlsConnectionFeature)features;
+                var clientCertPem = apiGatewayRequest?.RequestContext?.Authentication?.ClientCert?.ClientCertPem;
+                if (clientCertPem != null)
```

Comment:
Can you add a verification that the clientCertPem starts with **--------BEGIN CERTIFICATE-----** and **-----END CERTIFICATE-----**? If it doesn't then throw an meaningful error message that the certificate is invalid?

Reply:
Added a static method to Internal/Utilities.cs that does this check and the PEM to X509Certificate2 conversion. Updated bit places it's used.